### PR TITLE
Ignore overprovisioning in metadata protocol

### PR DESCRIPTION
**Pull request by adepue**
## Description

Issue # 

---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and Travis.CI is passing.

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: azurelinuxagent/ga/exthandlers.py [hunk 1]
```diff
@@ -315,7 +316,7 @@ def handle_ext_handlers(self, etag=None):
             logger.verbose("No extension handler config found")
             return
 
-        if conf.get_enable_overprovisioning():
+        if conf.get_enable_overprovisioning() and not isinstance(self.protocol, MetadataProtocol):
```
Comment (narrieta) [inline azurelinuxagent/ga/exthandlers.py]: maybe instead add a supports_overprovisioning() method to the protocol class? and a warning if the setting is true but the protocol doesn't support it? thanks

Comment (adepue) [reply azurelinuxagent/ga/exthandlers.py]: @narrieta Just updated with two commits to cover this.  Let me know what you think.
I debated having the base class raise NotImplemented, but didn't want to have to go to every subclass and override it.  Instead opted for the "everyone supports unless explicitly setting not".  Happy to change it if you prefer

Comment (narrieta) [reply azurelinuxagent/ga/exthandlers.py]: thanks @adepue, I think your approach is fine. can add a warning if the setting is true, but the protocol doesn't support it?

After that i can run our automation on your PR. On your side, there is a makepkg.py script on the root of the repo that will create a package for the agent. You can test that package to see if there are any other issues (you need to sudo to execute the script).

Comment (adepue) [reply azurelinuxagent/ga/exthandlers.py]: @narrieta Updated to log a warning.  Will test the package on this side as well.

File: azurelinuxagent/ga/exthandlers.py [hunk 2]
```diff
@@ -316,10 +316,13 @@ def handle_ext_handlers(self, etag=None):
             return
 
         if conf.get_enable_overprovisioning():
-            artifacts_profile = self.protocol.get_artifacts_profile()
-            if artifacts_profile and artifacts_profile.is_on_hold():
-                logger.info("Extension handling is on hold")
-                return
+            if not self.protocol.supports_overprovisioning():
+                logger.warn("Overprovisioning is enabled but protocol does not support it.")
```
Comment (narrieta) [inline azurelinuxagent/ga/exthandlers.py]: Sorry, my bad! I just realized this is in a loop that executes too frequently so the logs on Azure Stack would be too verbose. Can we change warn() to verbose()?

Comment (narrieta) [reply azurelinuxagent/ga/exthandlers.py]: Thank you, and sorry for the extra change!

**Comment by narrieta** [comment]
code looks good; running automation now

**Pull request merged**